src/libethash/ethash.h

```diff
@@ -74,57 +84,110 @@ uint64_t ethash_get_cachesize(const uint32_t block_number);
-{
-    ethash_cache c;
-    c.mem = (void *) cache;
-    ethash_compute_full_data(full, params, &c);
-}
+/**
+ * Allocate and initialize a new ethash_cache object
+ *
+ * @param params    The parameters to initialize it with. We are interested in
+ *                  the cache_size from here
+ * @param seed      Block seedhash to be used during the computation of the
+ *                  cache nodes
+ * @return          Newly allocated ethash_cache on success or NULL in case of
+ *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
+ */
+ethash_cache *ethash_cache_new(ethash_params const *params, ethash_h256_t const *seed);
+/**
+ * Frees a previously allocated ethash_cache
+ * @param c        The object to free
+ */
+void ethash_cache_delete(ethash_cache *c);
+bool ethash_compute_full_data(void *mem, ethash_params const *params, ethash_cache const *cache);
+
+/**
+ * Allocate and initialize a new ethash_light handler
+ *
+ * @param params    The parameters to initialize it with. We are interested in
+ *                  the cache_size from here
+ * @param seed      Block seedhash to be used during the computation of the
+ *                  cache nodes
+ * @return          Newly allocated ethash_light handler or NULL in case of
+ *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_cache_nodes()
+ */
+ethash_light_t ethash_light_new(ethash_params const *params, ethash_h256_t const *seed);
+/**
+ * Frees a previously allocated ethash_light handler
+ * @param light    The light handler to free
+ */
+void ethash_light_delete(ethash_light_t light);
+/**
+ * Calculate the light client data
+ *
+ * @param ret            An object of ethash_return_value to hold the return value
+ * @param light          The light client handler
+ * @param params         The parameters to use
+ * @param header_hash    The header hash to pack into the mix
+ * @param nonce          The nonce to pack into the mix
+ * @return               true if all went well and false if there were invalid
+ *                       parameters given.
+ */
+bool ethash_light_compute(ethash_return_value *ret,
+                          ethash_light_t light,
+                          ethash_params const *params,
+                          const ethash_h256_t *header_hash,
+                          const uint64_t nonce);
+/**
+ * Allocate and initialize a new ethash_full handler
+ *
+ * @param params    The parameters to initialize it with. We are interested in
+ *                  the full_size from here
+ * @param cache     A cache object to use that was allocated with @ref ethash_cache_new().
+ *                  Iff this function succeeds the ethash_full_t will take memory
+ *                  ownership of the cache and free it at deletion. If not then the user
+ *                  still has to handle freeing of the cache himself.
+ * @param seed      Block seedhash. TODO: Do we really need this in this function?
```

Comment:
@gavofyork Do we need the seed as a parameter here? In `ethash_light_new()` it's used to create the cache. Here it's not really used at all since the cache is given as an argument.